src/authorizations/components/redesigned/ResourceAccordionBody.tsx

```diff
@@ -0,0 +1,90 @@
+// Libraries
+import React, {FC} from 'react'
+
+// Clockface
+import {
+  Accordion,
+  FlexBox,
+  ComponentSize,
+  JustifyContent,
+  FlexDirection,
+  AlignItems,
+  InputLabel,
+  Toggle,
+  InputToggleType,
+} from '@influxdata/clockface'
+import {PermissionType} from 'src/types/tokens'
+
+interface Props {
+  resourceName: string
+  permissions: any
+  onToggle: (name, id, permission) => void
+  title: string
+}
+
+export const ResourceAccordionBody: FC<Props> = props => {
+  const {resourceName, permissions, onToggle, title} = props
+
+  const handleReadToggle = id => {
+    onToggle(resourceName, id, PermissionType.Read)
+  }
+  const handleWriteToggle = id => {
+    onToggle(resourceName, id, PermissionType.Write)
+  }
+
+  const accordionBody = telegraf => (
```

Comment:
can this function definition be pulled outside of the component so that it's only defined once, but called on every render? I don't think it needs to be re-defined every render call, right?

edit: looks like it depends on `handleReadToggle` and `handleWriteToggle` which depend on `resourceName`, so if those two functions are changed to have `id` *and* `resourceName` be arguments, I think we can pull all three function definitions out of the render function.

i don't even know how much more efficient is is, but it seems like it couldn't hurt

Reply:
Good question, I think `handleReadToggle` and `handleWriteToggle` are onChange handlers, which means it only provides with the `value` of the toggle when called. I will still need handlers in this component to get the value of toggle clicked, then I can execute the functions that we moved out with `id` and `resourceName`. What do you think?

Reply:
it's up to you. I think this is a micro-optimization at this point, more just a general question on how to do things